### PR TITLE
add draft post on fulltext story of fetching full text

### PR DESCRIPTION
**Pull request by sckott**
Let me know if you think anything could be changed to the structure, or any details that should be included or not included. Does it make sense? Does anything need to be explained better?

### CODE REVIEW

**Comment by stefaniebutland** [comment]
Thanks Scott. @steffilazerte will do first review and I'll check after. Can publish 2020-11-17

**Comment by sckott** [comment]
thanks. do I change folder name date now or later?

**Comment by stefaniebutland** [comment]
later is fine

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+title: "fulltext: behind the scenes"
+date: 2020-11-10
+author:
+  - Scott Chamberlain
+description: Behind the scenes of getting article full text.
+tags:
+  - literature
+  - textmining
+  - fulltext
+  - R
```
Comment (steffilazerte) [inline content/technotes/2020-11-10-fulltext-story/index.md:12]: What do you think?
```suggestion
  - R
  - packages
```

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+slug: "fulltext-story"
+title: "fulltext: behind the scenes"
```
Comment (steffilazerte) [inline content/technotes/2020-11-10-fulltext-story/index.md:3]: Title case
```suggestion
title: "fulltext: Behind the Scenes"
```

**Review by steffilazerte — COMMENTED** [review]
Hi @sckott,
This is a nice concise little post on the background of fulltext. It's a lot more complicated than I would have thought (but I'm not sure why I'm surprised! References are always messy). 

I have a couple of stylistic suggestions, typo fixes and two requests for more information (one where I couldn't follow the logic, and one to conclude the post just a bit). 

As always, with the exception of typos, title case and publication date, feel free to accept, alter or ignore any or all of my suggestions!

Thanks!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [x] title is in Title Case
* [ ] publication date is ok
  - **Just a reminder to update YAML and folder date to 2020-11-17**
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok

**Comment by sckott** [comment]
Thanks @steffilazerte for the comments, will have a look

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - literature
+  - textmining
+  - fulltext
+  - R
+output: 
+  html_document:
+    keep_md: true
+---
+
+[fulltext][] is a package I maintain for text-mining the scholarly literature ([package docs][ftdocs]). You can search for articles, fetch article metadata and abstracts, and fetch full text of some articles. Text-mining the scholarly literature is a research tool used ascross disciplines, and full text of articles is the gold standard in most cases.
```
[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md:18]: the changes in the last sentence change the meaning, i'll push a change incorporating your other fixes here though

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Some bad links: a new approach
+
+Given the problems with Crossref links to full text, I decided to work on another solution. I decided to make a web API of my own. [The API][ftdoiapi] was available at `ftdoi.org`, and all it did was accept a DOI for an article, then look up the Crossref publisher member ID, then use rules I matained for figuring out full text links to articles per publisher or journal, and return links to full text. 
+
+This solution worked pretty well, and had the added benefit that I could look at the API logs to see what publishers or DOIs users were most interested in getting links for - then I could work on making new mappings for those publishers/journals.
+
+However, essentially the only API users were R users. This meant that fulltext users were waiting for an extra http request that could fail or be slow if their internet connection is bad/slow or if I had a problem on the server for the API.
```
[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md:34]: will do

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Simplify and barriers
+
+I decided to retire the ftdoi.org API and do the mapping (DOI to Crossref member to full text link) inside of R. The first attempt at this was to implement the mapping in a separate package: [ftdoi][ftdoipkg]. I was about to submit to CRAN but then realized that new package submissions to CRAN can take a very, very long time, with no upper limit. Given that I wanted to wrap up the change away from the web API to the R side rather quickyly, I decided to pivot away from a separate package. Instead of a separate package, I simply moved the code I had in ftdoi to fulltext package. Then submitted a new version and done.
+
+Submitting a separate package really was the right decision from a software perspective as it was a ditinct set of code with a solid use case. However, given the unknown and possibly very long acceptance time on CRAN, folding the code into a package where I only had to submit a new version made more sense. Luckily new versions to CRAN are partly automated, so things go more smoothly and quickly. I definitely regret bloating the codebase of the fulltext package, but from a "getting things done" perspective it just made more sense.
+
```
[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md:41]: will do

[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md:46]: added a wrap up section

File: content/technotes/2020-11-10-fulltext-story/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md]: updated this section

File: content/technotes/2020-11-10-fulltext-story/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sckott) [reply content/technotes/2020-11-10-fulltext-story/index.md]: updated this sectoin

**Comment by sckott** [comment]
once we're done with changes, i'll make the change to the folder name

File: content/technotes/2020-11-10-fulltext-story/index.md [hunk 6]
```diff
@@ -10,36 +10,43 @@ tags:
...
 
-Submitting a separate package really was the right decision from a software perspective as it was a ditinct set of code with a solid use case. However, given the unknown and possibly very long acceptance time on CRAN, folding the code into a package where I only had to submit a new version made more sense. Luckily new versions to CRAN are partly automated, so things go more smoothly and quickly. I definitely regret bloating the codebase of the fulltext package, but from a "getting things done" perspective it just made more sense.
+## Onward
 
+Moving forward there will be improvements in fetching full text of articles in fulltext package as we have time to make mappings on a publisher and/or journal basis. These improvements will require new versions of fulltext to get to CRAN. When we used the ftdoi.org API users could benefit from new journal/publisher mappings as soon as the API was updated, which is very fast - so addition of new mappings will take longer now assuming users only install CRAN versions.
```
Comment (steffilazerte) [inline content/technotes/2020-11-10-fulltext-story/index.md:47]: I like the wrap up section, just one last suggested clarification!
```suggestion
Moving forward there will be improvements in fetching full text of articles in the fulltext package as we make mappings on a publisher and/or journal basis. Unfortunately these improvements require new versions of fulltext to get to CRAN. When we used the ftdoi.org API users could benefit from new journal/publisher mappings as soon as the API was updated, which is very fast - but the addition of new mappings will take longer now assuming users only install CRAN versions.
```

**Comment by steffilazerte** [comment]
I'm happy! Thanks for looking into this so quickly. @stefaniebutland you're good to go!

**Comment by sckott** [comment]
thanks @steffilazerte !

**Comment by steffilazerte** [comment]
@stefaniebutland Are you doing a last check before we publish tomorrow?

@sckott Comments or suggestions for the tweet?

---

[technote] fulltext: Behind the Scenes

From custom APIs to internal mappings, @sckottie shares a behind-the-scenes look at the evolution of how the package obtains fulltext articles.

https://ropensci.org/technotes/2020/11/17/fulltext-story/

#rstats #fulltext

**Comment by sckott** [comment]
i think just change `obtains fulltext articles` to `obtains full-text articles`

**Comment by steffilazerte** [comment]
Great!

---

[technote] fulltext: Behind the Scenes

From custom APIs to internal mappings, @sckottie shares a behind-the-scenes look at the evolution of how the package obtains full-text articles.

https://ropensci.org/technotes/2020/11/17/fulltext-story/

#rstats #fulltext

**Comment by stefaniebutland** [comment]
Nice post and the tweet gives a good idea of that value that's in the post!!

thanks you two.

**Review by stefaniebutland — APPROVED** [review]

**Pull request merged**